Updated Service discovery (Consul service discovery) for Choreo Connect (#3615)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md

# Configure Choreo Connect with Consul service mesh deployed on Kubernetes

> Example:

Reviewer: Please add some kind of explanation or overview that covers the following:

- What are we trying to achieve with this documentation.
- Description of what Consul service mesh is.
- Why someone would want to connect Choreo Connect with Consul.

> ![service discovery](../../../../assets/img/deploy/consul-reference-k8s.png)

Reviewer: ```suggestion
![service discovery]({{base_path}}/assets/img/deploy/consul-reference-k8s.png)
```
Always use base_path for links.

## Step 1 - Deploy Consul on K8s
Deploy Consul using the [official Helm Chart](https://helm.releases.hashicorp.com/).<br>
For more instructions, see the [Consul official documentation](https://www.consul.io/docs/k8s).
## Step 2 - Modify Choreo Connect k8s artifacts 

> Add the `'consul.hashicorp.com/connect-inject': 'true'` annotation to Adapter's deployment spec

Reviewer: ```suggestion
Add the `'consul.hashicorp.com/connect-inject': 'true'` annotation to the Adapter's deployment specification.
```
Shall we add which file exactly we are meant to modify here?

Author: fixed 364f6cb

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: choreo-connect-adapter
spec:
  replicas: 1
  selector:
    matchLabels:
      app: choreo-connect-adapter
  template:
    metadata:
      labels:
        app: choreo-connect-adapter
      annotations:
        'consul.hashicorp.com/connect-inject': 'true'
```

> In the config.toml config map, go to the \[adapter.consul] section and update it's `url` with k8s node ip.

Reviewer: ```suggestion
In the config.toml config file, go to the \[adapter.consul] section and update the `url` with the IP address of the Kubernetes node.
```
